app/src/main/java/com/iic/mokojin/CurrentMatchFragment.java

```diff
@@ -0,0 +1,186 @@
+
+        return rootView;
+    }
+
+    private void refreshCurrentMatch() {
+        mCurrentMatch = null;
+        Match.getCurrent().continueWith(new Continuation<Match, Void>() {
+            @Override
+            public Void then(Task<Match> task) throws Exception {
+                if (task.isCancelled()) {
+                    Log.d(LOG_TAG, "Fetching of current match was cancelled");
+                } else if (task.isFaulted()) {
+                    Log.e(LOG_TAG, "Error fetching current match", task.getError());
+                } else {
+                    mCurrentMatch = task.getResult();
+                    refreshUI();
+                }
+                return null;
+            }
+        }, Task.UI_THREAD_EXECUTOR);
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        refreshCurrentMatch();
+    }
+
+    private void displayCharacter(int characterId, ImageView imageView) {
+        int resourceId = getResources().getIdentifier("player_" + characterId, "drawable", this.getActivity().getPackageName());
+        Drawable charDrawable = getResources().getDrawable(resourceId);
+        imageView.setImageDrawable(charDrawable);
+    }
+
+    private void hideBackImage(ImageView frontImageView, ImageView backImageView) {
+        backImageView.setVisibility(View.GONE);
+
+        // Get rid of margins and put it in the middle
+        FrameLayout.LayoutParams layoutParams = (FrameLayout.LayoutParams)frontImageView.getLayoutParams();
+        layoutParams.rightMargin = 0;
+        layoutParams.topMargin = 0;
+        layoutParams.gravity = Gravity.CENTER;
+        frontImageView.setLayoutParams(layoutParams);
+    }
+
+    private void showBackImage(ImageView frontImageView, ImageView backImageView) {
+        backImageView.setVisibility(View.VISIBLE);
+
+        FrameLayout.LayoutParams layoutParams = (FrameLayout.LayoutParams)frontImageView.getLayoutParams();
+        layoutParams.rightMargin = getResources().getDimensionPixelSize(R.dimen.character_separation_amount);
+        layoutParams.topMargin = getResources().getDimensionPixelSize(R.dimen.character_separation_amount);;
+        layoutParams.gravity = Gravity.NO_GRAVITY;
+        frontImageView.setLayoutParams(layoutParams);
+    }
+
+    private void setPlayerImages(Player player, ImageView charA, ImageView charB) {
+        if (player.getCharacterA() != null && player.getCharacterB() != null) {
+            displayCharacter(player.getCharacterA().getCharacterId(), charA);
+            displayCharacter(player.getCharacterB().getCharacterId(), charB);
+            showBackImage(charA, charB);
+        } else if (player.getCharacterA() != null) {
+            displayCharacter(player.getCharacterA().getCharacterId(), charA);
+            hideBackImage(charA, charB);
+        } else {
+            charA.setImageDrawable(getResources().getDrawable(R.drawable.player_empty));
+            hideBackImage(charA, charB);
+        }
```

Comment:
I feel like we can always set the images for both ImageViews and just make sure to have a graceful fallback when one character doesn't exist.